pilot/pkg/model/validation.go

```diff
@@ -322,6 +322,19 @@ func ValidatePercent(val int32) error {
 	return nil
 }
 
+// validatePercentageOrDefault checks if the specified fractional percentage is
+// valid. If a nil fractional percentage is supplied, it validates the default
+// integer percent value.
+func validatePercentageOrDefault(percentage *networking.Percent, defaultPercent int32) error {
+	if percentage != nil {
+		if percentage.Value < 0.0 || percentage.Value > 100.0 || (percentage.Value > 0.0 && percentage.Value < 0.0001) {
```

Comment:
why the 0.001 limitation ? We can always multiply by 1M right? Oh its the JSON precision thing?

Reply:
This isn't because of JSON precision. It's because we'd use a 1,000,000 denominator when we multiply the numerator with 10,000. Correct me if I'm wrong.

Reply:
is there something wrong with that? We can always use the 1M denominator right?

Reply:
Yes, that's what we're doing, but we cannot go beyond that i.e. the largest value we can multiply the numerator with is 10,000 (when the denominator is 1M). Isn't it?